### DISCUSSION TITLE
LIDAR Navarra - Procesado 2º Cobertura

### DISCUSSION THREAD

**Discussion by manudona92** [manudona92/MapantES#50] [General]
Navarra es el el ganador en cuanto a "Lote más difícil". Tiene 14 pts/m2 la 2º cobertura. La 1º cobertura tiene resultados un poco pobres en la clasificación por lo que utilizar la 2º cobertura, si sale bien, puede tener mejores resultados (mejor definición de la densidad y cortados también). 

El parámetro que estoy testeando se llamará "Navarra_5.ini". Lo subiré cuando lo valide. 

Cobertura de Navarra = 2º Cobertura , 2017

Workflow que he estado probando. 

1) las2las **incluir aquí recortar a tiles de 200x200 m** -keep_classification 2 3 4 5 (este paso se puede hacer en lasthin directamente) -olaz

Esto se hace porque Navarra tiene muchisimo ruido por las características del sensor. Quedándonos solo con esas clases evitamos trabajar con nubes de ruido (ej. Carballo) y reducir el tamaño del LAZ considerablemente 

2) lasthin  -i  -ignore_class 3 4 5 -keep_classification 2 3 4 5 -step 1 -percentile 50 -olaz

Con esto reducimos el ruido a borde de suelo y ver un excesivo número de mini cortados.

3) lasnoise -ignore_class 3 4 5 -step_xy 2 -step_z 1 -remove_noise -olaz

Para "acabar" con ese ruido aéreo/subterráneo de suelo. Alguno quedará pero ayudará. 

Es laborioso, habría que añadir un paso que es lasthin. En los test que he hecho no me ha dado ningún tipo de deformación y tarda muy poco (como lasnoise). Creo que este lote es especial y lo podríamos procesar a parte, internamente en un grupo reducido o de otra forma habria que obligar a la gente a meterse lasthin en su carpeta. Lo podríamos dejar para procesar después de Cataluña, Madrid, CV, Murcia, La Rioja.

Un ejemplo de 1º vs 2º cobertura
![image](https://user-images.githubusercontent.com/76484741/111683073-5e2b5e00-8825-11eb-8933-92df61ee9b25.png)

**Comment by agustincaballero** [comment]
Lo veo bien, lo único es que habría que calibrar bien si 200x200 metros es un tamaño adecuado o si se pueden hacer más grandes. Hay que pensar que si hacemos 200x200, saldrían 25 veces más cuadrículas por kilómetro cuadrado que en Galicia o Canarias.
¿Qué tamaño medio aproximado tiene una cuadrículas 2x2km de Navarra frente a otras, por ejemplo las de Galicia?
Lo del lasthin no lo veo ningún problema, sería simplemente meter el lasthin en el fichero zip que ya se tienen que descargar al instalar el cliente. El cliente avisaría si falta ese fichero antes de hacer nada.

**Reply by manudona92** [reply]
Correcto. En las últimas pruebas con tiles de 250x250 m, éstos pesaban entre 5 y 20 mb/tile. 

Un tile de 1x1km (así es como se pueden descargar en el CNIG, no estan en 2x2 km), pesan entre 500-800 mb. Con el las2las (quitar las clases que no nos interesan) y lasthin, este tamaño se puede reducir hasta un 20-50%.  Mi primera opción sería procesarlos con 250x250 m, donde el tamaño de tile sería suficientemente asequible para todo tipo de máquina. Otra prueba sería irnos a 500x500 m (30-100 mb/tile).

**Comment by manudona92** [comment]
Tiempos de procesamiento test:


1)  Intel Core 2 quad Q9300 2.50GHz, con 3GB de RAM.

- 1x1 km: **41'**
- 500x500 m (4 inst) **26'**
- 250x250 m (4 inst) **41'**

2) Intel Core i7-2670QM 2.20GHz, 4GB RAM

- 1x1 km **34'**
- 500x500 m (4 inst)  **30'**
- 250x250 m (4 inst)  **38'**

3)  Intel Core i5-4440 3.10GHz, con 12GB de RAM.

- 1x1 km: **20'**
- 500x500 m (4 inst) **9'**
- 250x250 m (4 inst) **17'**

Tamaño de fichero LAZ in: 216 mb
Tamaño de ficheros de salida: 

- 1x1 km 1.21 GB
- 500x500 m 3.59 GB
- 250x250 m 6.36 GB
